scripts/release.sh

```diff
@@ -349,6 +349,7 @@ function parse_flags() {
       --dot-release) is_dot_release=1 ;;
       --auto-release) is_auto_release=1 ;;
       --from-latest-nightly) FROM_NIGHTLY_RELEASE=latest ;;
+      --use-ko-docker-repo) USE_KO_DOCKER_REPO=1 ;;
```

Comment:
My apologies, I should have read this PR more thoroughly. So I see that what you want is to be able to specify a custom location when you build your test release. So you don't need a new flag, you just need to call:

./hack/release.sh --release-gcr gcr.io/my-gcr --release-gcs my-gcs --notag-release --publish

You can add --skip-tests if you don't want to run all tests (unit+e2e) before building the release.

Reply:
what do I put for "my-gcs"?  When I replaced the --release-gcr flag with my private repo and left "my-gcs" as is, it didn't seem to do anything, no build, no push.  And yet it also didn't give an error.

Reply:
my-gcs is any GCS bucket you have write access to. From serving HEAD, if I run the command above (with --skip-tests) I do get the images published to my GCR and the manifests copied to my bucket. In the output, I see the banners "Building the release" (and the expected output from ko) and also "New release published successfully" at the end. I think we're using different scripts. Let's debug this offline on Slack.

Reply:
I'm not using GCS - that's part of the point of this PR. We need to allow people to build Knative outside of Google.

Reply:
Ideally, I think it would be nice if scripts like this took two input optional parameters:
1 - the registry in which to push the build images (e.g. KO_DOCKER_REPO, default to local docker)
2 - the dir in which to place all files needed to do an install (default to some `install_files` type of dir).

That gives people who want to build knative themselves the easiest path to getting all of the information they need in very well defined (local) locations, with minimal effort.

If after that some wrapper script wanted to push the files to some company specific location (like GCS) they can do so, but I think the core script should be generic and easy for everyone to run.

I'm not against this script knowing how to talk to company specific components (like GCS) but that should require a flag to do so - the default should be the easiest... store the files in a dir.

Reply:
Please note that this script was created to publish Knative, and that currently relies on GCP. That said, I understand and agree that we can add features to make the script store the manifests locally instead of a GCS bucket, that's simple.

Reply:
https://github.com/knative/test-infra/pull/1554 should give you the feature you want.